ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -328,12 +333,32 @@ void OwInterface::managePanTilt (const string& opname,
 void OwInterface::cameraCallback (const sensor_msgs::Image::ConstPtr& msg)
 {
   // NOTE: the received image is ignored for now.
-
+  m_pointCloudRecieved = false;
   if (operationRunning (Op_TakePicture)) {
+    ros::Rate rate(10);
+    int timeout = 0;
+    // We wait for the pointcloud as well or the 5 sec timeout before marking as finished
+    while(m_pointCloudRecieved == false && timeout < 50){
```

Comment:
Best to make a named constant for the timeout (50) which can be defined local to this file.   I don't have a suggestion for a better approach than timeout -- this seems reasonable.